packages/flutter_tools/lib/src/windows/visual_studio.dart

```diff
@@ -167,6 +171,29 @@ class VisualStudio {
     ]);
   }
 
+  String? getCmakeVersion() {
+    final String _cmakeDocPath = _fileSystem.path.joinAll(<String>[
+      _usableVisualStudioDetails[_installationPathKey] as String,
+      'Common7',
+      'IDE',
+      'CommonExtensions',
+      'Microsoft',
+      'CMake',
+      'CMake',
+      'doc',
+    ]);
+    final Directory _cmakeDocDir = globals.fs.directory(_cmakeDocPath);
+      if (!_cmakeDocDir.existsSync()) {
+        return null;
+    }
+    for (final Directory dir in _cmakeDocDir.listSync().whereType<Directory>()) {
```

Comment:
This seems like a convoluted way of getting the version; it unnecessarily adds more dependencies on internal details of install layout, and it's not clear to me that there's a guarantee that this will always give the correct answer (e.g., is it possible for a user to have multiple entries here if an older VS installation wasn't fully cleaned up?)

If we're going to use the CMake version as the gate, then we should run `cmake --version`. It uses the path, which we already have, and gives a definitive answer about what version will run.

Reply:
This same approach is used in other areas in `flutter_tools` for instance [getting Android Studio version on Windows](https://github.com/flutter/flutter/blob/ac05e7e3fefd63d39c73c44b8097927b673ce942/packages/flutter_tools/lib/src/android/android_studio.dart#L374-L379). I've just replicated that for the CMake directory. 

I was worried about different versions too but VS installer doesn't provide multiple CMake versions, here I can only install `3.20`. I also tried installing `individual components` in the installer, only one CMake shows up in the result without providing an ability to install older CMake from VS.

Reply:
> This same approach is used in other areas in `flutter_tools` for instance [getting Android Studio version on Windows](https://github.com/flutter/flutter/blob/ac05e7e3fefd63d39c73c44b8097927b673ce942/packages/flutter_tools/lib/src/android/android_studio.dart#L374). I've just replicated that for the CMake directory.

There's a significant difference between "is the installation folder present in the standard system install location for apps" and "is a documentation folder present at a specific location in the internals of a VS install".

Again, we already have the path to the binary in question, and the binary has the ability to report its version. That approach is less fragile in several important regards. It's also exactly how we determine the `cmake` version on Linux.

I don't understand the pushback here; what are the arguments in favor of the approach you've done here?

> I was worried about different versions too but VS installer doesn't provide multiple CMake versions, here I can only install `3.20`. I also tried looking into `individual components` in the installer, only one CMake shows up in the result without providing an ability to install older CMake from VS.

That's not the scenario I described. You are not going to be able to prove the negative that there are no scenarios where a series of incremental VS upgrades might potentially have left an old documentation directory.